thorlcr/activities/thactivityutil.cpp

```diff
@@ -783,9 +783,42 @@ IFileIO *createMultipleWrite(CActivityBase *activity, IPartDescriptor &partDesc,
     Owned<IFileIO> fileio;
     if (compress)
     {
+        // rowdif used if recordSize > 0, fallback set to LZ4
+        unsigned __int64 compMethod = COMPRESS_METHOD_LZ4;
+        // fallback override
+        StringBuffer compType;
+        activity->getOpt(THOROPT_FBACK_COMPRESS_TYPE, compType);
```

Comment:
```
#define THOROPT_COMPRESS_SPILL_TYPE   "spillCompressorType"     // Compress spill type, e.g. LZW, FLZ, LZ4 or DEN                                (default = LZ4)
#define THOROPT_FBACK_COMPRESS_TYPE   "fallbackCompressorType"  // Compress type when recSize 0 or too small, e.g. LZW, FLZ, LZ4 or DEN          (default = LZ4)
```

It's not clear to me, why there are 2 options here, or why THOROPT_FBACK_COMPRESS_TYPE is used in this case.

Also, I think there's a problem here, if the this method (createMultipleWrite()) is being used to write a published compressed file, then there are compatibility problems..

For spills, which include transient inter-subgraph spills that use createMultipleWrite(), using the same THOROPT_COMPRESS_SPILL_TYPE option is appropriate. See thediskbaseslave.cpp and TDXtemporary helper flag that indicates temporary diskwrite.
For non-spills (published files) we should for the time being avoid changing the format.

So I think you also need change createMultipleWrite() and callers to pass in the compression type.